### PR TITLE
Modified Prometheus extension to support adding a prefix to metric names

### PR DESCRIPTION
**Pull request by topranks**
This change allows for a user-configurable prefix to be defined for the Prometheus extension.  If the configuration option is set the configured string will be prepended to all metric names exported to Prometheus Push Gateway.  If the option is not set it will default to an empty string, and metric names will be as they would have been otherwise, ensuring backwards compatibility.

See community post for more info:

https://community.librenms.org/t/prometheus-extenstion-add-option-to-prefix-metric-labels-with-a-user-configured-string/16973

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [x] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [ ] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by murrant** [comment]
Needs to be added to config_definitions.json

**Comment by topranks** [comment]
Thanks @murrant I hope I've done that correctly now.

File: misc/config_definitions.json [hunk 1]
```diff
@@ -4771,6 +4771,13 @@
             "section": "prometheus",
             "order": 3
         },
+	"prometheus.prefix": {
+	    "default": "",
+	    "type": "text",
+            "group": "poller",
```
Comment (murrant) [inline misc/config_definitions.json:4777]: Indentation is off and you'll need to add strings to resources/lang/en/settings.php

Comment (topranks) [reply misc/config_definitions.json:4777]: I believe I fixed the indentation in the subsequent commit.  Looking at settings.php now thanks :)

Comment (topranks) [reply misc/config_definitions.json:4777]: OK thanks for the assistance with this, I believe the latest commit also adds the required to resources/lang/en/settings.php

File: LibreNMS/Data/Store/Prometheus.php [hunk 2]
```diff
@@ -47,6 +47,10 @@ public function __construct(\GuzzleHttp\Client $client)
         $url = Config::get('prometheus.url');
         $job = Config::get('prometheus.job', 'librenms');
         $this->base_uri = "$url/metrics/job/$job/instance/";
+        $this->prefix = Config::get('prometheus.prefix', '');
```
[the remark this answers is not in the document]
Comment (topranks) [reply LibreNMS/Data/Store/Prometheus.php:51]: Thanks @murrant.  This is not proving to be my finest hour.  Latest commit has it, thank you for all the help.

[the remark this answers is not in the document]
Comment (murrant) [reply LibreNMS/Data/Store/Prometheus.php:51]: no worries, can't be perfect 100% of the time :D

**Review by murrant — APPROVED** [review]

**Comment by topranks** [comment]
Thanks for the approval and merge!  Not to mention the pointers :+1:

**Pull request merged**